Dockerfile

```diff
@@ -90,69 +154,63 @@ RUN set -eux; \
 	done; \
 	[ -n "$success" ]; \
 	\
-	gpg --batch --verify tomcat.tar.gz.asc tomcat.tar.gz; \
-	tar -xvf tomcat.tar.gz --strip-components=1; \
-	rm bin/*.bat; \
-	rm tomcat.tar.gz*; \
-	\
-	nativeBuildDir="$(mktemp -d)"; \
-	[ ${JAVA_MAJOR} == 8 ] && JAVA_MAJOR_PKG=1.8.0; \
-	JRE_PKG_VERSION=$(rpm -qa java-${JAVA_MAJOR_PKG:-$JAVA_MAJOR}-openjdk-headless --queryformat "%{RPMTAG_VERSION}"); \
-	tar -xvf bin/tomcat-native.tar.gz -C "$nativeBuildDir" --strip-components=1; \
-	[ ${DISTRIB_MAJOR} = 7 ] && nativeBuildDeps=" \
-		apr-devel-1.4.8-7.el7 \
-		gcc-4.8.5-44.el7 \
-		make-3.82-24.el7 \
-		openssl-devel-1.0.2k-22.el7_9 \
-	"; \
-	yum install -y "java-${JAVA_MAJOR_PKG:-$JAVA_MAJOR}-openjdk-devel-${JRE_PKG_VERSION}"; \
-	yum install -y $nativeBuildDeps; \
-	( \
-		export CATALINA_HOME="$PWD"; \
-		cd "$nativeBuildDir/native"; \
-		./configure \
-			--with-java-home="$JDK_HOME" \
-			--libdir="$TOMCAT_NATIVE_LIBDIR" \
-			--prefix="$CATALINA_HOME" \
-			--with-apr="$(which apr-1-config)" \
-			--with-ssl=yes; \
-		make -j "$(nproc)"; \
-		make install; \
-	); \
-	yum history -y rollback last-1; \
-	find /etc -mindepth 2 -name *.rpmsave -exec rm -v '{}' +; \
-	rm -rf "$nativeBuildDir"; \
-	rm bin/tomcat-native.tar.gz; \
-	\
+	gpg --batch --verify tomcat.tar.gz.asc tomcat.tar.gz && \
+	tar -zxf tomcat.tar.gz -C /build --strip-components=1
+
+WORKDIR /build
+RUN \
 # sh removes env vars it doesn't support (ones with periods)
 # https://github.com/docker-library/tomcat/issues/77
 	find ./bin/ -name '*.sh' -exec sed -ri 's|^#!/bin/sh$|#!/usr/bin/env bash|' '{}' +; \
 	\
 # fix permissions (especially for running as non-root)
 # https://github.com/docker-library/tomcat/issues/35
 	chmod -R +rX .; \
-	chmod 777 logs work ; \
+	chmod 330 logs work ; \
 	\
 	# Security improvements:
 	# Remove server banner, Turn off loggin by the VersionLoggerListener, enable remoteIP valve so we know who we're talking to
 	sed -i \
           -e "s/\  <Listener\ className=\"org.apache.catalina.startup.VersionLoggerListener\"/\  <Listener\ className=\"org.apache.catalina.startup.VersionLoggerListener\"\ logArgs=\"false\"/g" \
           -e "s%\(^\s*</Host>\)%\t<Valve className=\"org.apache.catalina.valves.RemoteIpValve\" />\n\n\1%" \
-          -e "s/\    <Connector\ port=\"8080\"\ protocol=\"HTTP\/1.1\"/\    <Connector\ port=\"8080\"\ protocol=\"HTTP\/1.1\"\n\               Server=\" \"/g" /usr/local/tomcat/conf/server.xml; \
+          -e "s/\    <Connector\ port=\"8080\"\ protocol=\"HTTP\/1.1\"/\    <Connector\ port=\"8080\"\ protocol=\"HTTP\/1.1\"\n\               Server=\" \"/g" conf/server.xml; \
```

Comment:
Out of curiosity, why we want to use relative path?

Reply:
2 things:
* multistage build makes the installtion path irrelevant (we're actually unzipping in /build/something) 
* we're using WORKDIR just before so we know that commands runs from within the tomcat distribution root dir)